.github/workflows/checks.yml

```
@@ -5,6 +5,33 @@ on:
   pull_request:
 
 jobs:
+  fmt:
+    name: Format
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install nightly@nightly toolchain
+      uses: dtolnay/rust-toolchain@nightly
+      with:
+        components: rustfmt
+    - name: Format
+      run: cargo +nightly fmt -- --check
+
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install stable@stable toolchain
+      uses: dtolnay/rust-toolchain@stable
+      with:
+        components: clippy
+    - uses: Swatinem/rust-cache@v2
+    - name: Run Clippy
+      run: cargo clippy --locked --all-features --all-targets
```

```suggestion
      run: cargo clippy --locked --all-features --all-targets -- -D warnings
```

Is `--all-features --all-targets` the same as `--all`?

---

From google AI: was actually hard to find reference to it 

The `--all` argument in Cargo typically extends the scope of a command to encompass all relevant components within a project or workspace, such as all packages, all features, or all benchmarks, depending on the command it's paired with. It is important to consult the documentation for the specific Cargo subcommand you are using to understand the precise effect of `--all` in that context.

When it first appeared I remember it being suggested as a replacement for `--worspace`,  so it also checks stuff benches and examples, while --all-targets is architecture and --all-features, well features

---

No need for `-- -D warnings`

```toml
[workspace.lints.clippy]
all = "deny"
```